onnxruntime/python/tools/kernel_explorer/CMakeLists.txt

```diff
@@ -0,0 +1,33 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+project(kernel_explorer)
+cmake_minimum_required(VERSION 3.21)
+
+include(../../../../cmake/external/pybind11.cmake)
+
+# Memory error with Release build (-o3 -DNDEBUG), AMD is currently investigating
+set(CMAKE_BUILD_TYPE Debug)
+set(CMAKE_CXX_COMPILER /opt/rocm/bin/hipcc)
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fpic")
+set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_CXX_EXTENSIONS OFF)
+
+list(APPEND CMAKE_PREFIX_PATH /opt/rocm/hip /opt/rocm)
+find_package(hip)
+find_package(PythonLibs 3.8 EXACT REQUIRED)
+
+include_directories(${PYTHON_INCLUDE_DIRS})
+include_directories(${pybind11_INCLUDE_DIRS})
+include_directories(.)
+
+add_library(timer timer.cpp)
+add_library(vector_add kernels/vector_add.cpp)
+add_library(kernel_explorer SHARED kernel_explorer.cpp)
+target_link_libraries(kernel_explorer timer vector_add ${PYTHON_LIBRARIES})
```

Comment:
besides test new kernels, i guess another interesting piece is whether we can use this tool to tune the configuration for existing ort gpu kernels, by ort gpu kernels i mean the device kernel (most in *.cu file), not the ort "OpKernel", so you don't need to worried about the signature. Today all the ort gpu kernels are launched with same configuration, it will be interesting if we can turn a better configuration without change the GPU code.

for example, when build this cmake, if we can configure the kernel path to an existing ort gpu kernel, auto-generate the python binding, and run the tuning tool with different block/grid size, that will be more useful, compare with manually written new kernels here.

Reply:
Yes, it will take us a few iterations to get there, but that's definitely the direction we are going.